### PR TITLE
Improve content app performance on head requests

### LINKED ISSUE #2924: Content app is making queries on head requests
```
(pulp) [vagrant@pulp3-source-centos8-stream-fips ~]$ nc 127.0.0.1 4444                                                                                                                   [8/8]
> /home/vagrant/devel/pulpcore/pulpcore/content/handler.py(144)list_distributions()                                                                                                           
-> return HTTPOk(headers={"Content-Type": "text/html"}, body=self.render_html(directory_list))                                                                                                
(Pdb) directory_list                                                                                                                                                                          
['58016a04-8204-4ced-8f87-5318716e525d/']                                                                                                                                                     
(Pdb) base_paths                                                                                                                                                                              
['58016a04-8204-4ced-8f87-5318716e525d']            
(Pdb) request.method
'HEAD'
(Pdb) c
```

### PR DESCRIPTION
**Pull request by fao89**
closes #2924

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/content/handler.py [hunk 1]
```diff
@@ -138,6 +138,8 @@ def get_base_paths_blocking():
                 )
             return base_paths
 
+        if request.method.lower() == "head":
+            return HTTPOk(headers={"Content-Type": "text/html"})
```
Comment (bmbouter) [inline pulpcore/content/handler.py:142]: Does it even need the header specified?

**Review by bmbouter — CHANGES_REQUESTED** [review]
Would you be willing to write some docs on how users can check if a specific content app is alive by issuing a `HEAD /pulp/content/` call (or whatever [CONTENT_PATH_PREFIX](https://docs.pulpproject.org/pulpcore/configuration/settings.html#content-path-prefix) is set to).

I think it would make sense as a new section called "Availability" in this area https://docs.pulpproject.org/pulpcore/components.html (or wherever it makes sense to you).

**Review by bmbouter — APPROVED** [review]
Thank you!

**Comment by bmbouter** [comment]
FWIW I don't think this is worth backporting

**Review by mdellweg — APPROVED** [review]

**Pull request merged**